rinja_derive/src/generator.rs

```diff
@@ -161,12 +165,12 @@ impl<'a, 'h> Generator<'a, 'h> {
 
         let size_hint = self.impl_template_inner(ctx, buf)?;
 
-        buf.write(format_args!(
-            "\
-                rinja::Result::Ok(())\
-            }}\
-            const SIZE_HINT: rinja::helpers::core::primitive::usize = {size_hint}usize;",
-        ));
+        buf.write("rinja::Result::Ok(()) }");
+        if tmpl_kind == TmplKind::Struct {
+            buf.write(format_args!(
+                "const SIZE_HINT: rinja::helpers::core::primitive::usize = {size_hint}usize;",
```

Comment:
Seems a bit strange that the number is marked as usize twice. Why not:

```suggestion
                "const SIZE_HINT: rinja::helpers::core::primitive::usize = {size_hint};",
```

?

Reply:
Consistency: `quote!`'d numbers append the type. E.g. the parser tests use `quote!` and would fail otherwise.

Reply:
Makes sense. :+1